pkg/controllers/provisioning/scheduling/nodeclaimtemplate.go

```diff
@@ -82,6 +82,7 @@ func (i *NodeClaimTemplate) ToNodeClaim(nodePool *v1beta1.NodePool) *v1beta1.Nod
 		},
 		Spec: i.Spec,
 	}
+	nc.Spec.StartupTaints = append(nc.Spec.StartupTaints, v1beta1.UnregisteredNoExecuteTaint)
```

Comment:
Same comment here, I think we should avoid adding this as an actual startup taint on the NodeClaim, and this being injected into the Node userData by the CloudProvider should be implicit to the implementation

Reply:
Done.